nixos/modules/config/xkb.nix

```diff
@@ -57,17 +57,29 @@ in
     };
   };
 
-  config = {
-    system.checks = lib.singleton (pkgs.runCommand "xkb-validated" {
-      inherit (cfg) dir model layout variant options;
-      nativeBuildInputs = with pkgs.buildPackages; [ xkbvalidate ];
-      preferLocalBuild = true;
-    } ''
-      ${lib.optionalString (config.environment.sessionVariables ? XKB_CONFIG_ROOT)
-        "export XKB_CONFIG_ROOT=${config.environment.sessionVariables.XKB_CONFIG_ROOT}"
-      }
-      XKB_CONFIG_ROOT="$dir" xkbvalidate "$model" "$layout" "$variant" "$options"
-      touch "$out"
-    '');
+  config = lib.mkIf cfg.enable {
+    environment.etc."X11/xkb".source = cfg.dir;
+
+    system.checks = lib.singleton (
+      pkgs.runCommand "xkb-validated"
+        {
+          inherit (cfg)
+            dir
+            model
+            layout
+            variant
+            options
+            ;
+          nativeBuildInputs = with pkgs.buildPackages; [ xkbvalidate ];
+          preferLocalBuild = true;
+        }
+        ''
+          ${lib.optionalString (
+            config.environment.sessionVariables ? XKB_CONFIG_ROOT
+          ) "export XKB_CONFIG_ROOT=${config.environment.sessionVariables.XKB_CONFIG_ROOT}"}
+          XKB_CONFIG_ROOT="$dir" xkbvalidate "$model" "$layout" "$variant" "$options"
+          touch "$out"
+        ''
+    );
```

Comment:
It would have been nice if this change which mostly includes indentation / formatting changes, would have been split to two commits, or perhaps even 3 - since the `lib.mkIf cfg.enable` part is also a real change, in contrast to the rest of the `system.checks` formatting changes.

Reply:
This PR is going to get less and less commits the more often I rebase it because it is super hard to keep track of all of those commits.